### PR TITLE
spdlog: update to version 1.4.2

### PR DESCRIPTION
**Pull request by XNephila**
#### Description

Update spdlog lib to 1.4.2

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.2
Xcode 8.x

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by reneeotten** [comment]
the build seems to fail with a checksum error - can you please check?

**Comment by XNephila** [comment]
will be fixed in short time.

**Comment by cjones051073** [comment]
@XNephila ping. Can you address the checksum errors ?

**Comment by XNephila** [comment]
Sorry for the delay, I guess it's ready to merge.
Thanks!

File: devel/spdlog/Portfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ PortSystem          1.0
 PortGroup           github 1.0
 PortGroup           cmake  1.1
 
-github.setup        gabime spdlog 0.16.3 v
+github.setup        gabime spdlog 1.4.2 v
 #revision            0
```
Comment (mf2k) [inline devel/spdlog/Portfile]: Don't comment out the line. Leave the `revision` in at 0.

Comment (reneeotten) [reply devel/spdlog/Portfile]: when doing that, also please squash your commits into one

**Comment by XNephila** [comment]
I didn't find a way to squash the commits directly from Github website, should I create a new branch or do it by gitbash?

**Comment by reneeotten** [comment]
also the build fails with Xcode 7.3 with the message:
```
    static thread_local const size_t tid = _thread_id();
           ^
1 error generated.
```

Ideally, that should be resolved by adding ```compiler.thread_local_storage yes``` to the Portfile. Or if that doesn't work you might need to explicitly blacklist clang < 800 for the time being; in that case, please add a comment like below as well.
```
# setting "compiler.thread_local_storage yes" does not work, so for now just use blacklisting
# See: https://lists.macports.org/pipermail/macports-dev/2019-November/041503.html
compiler.blacklist-append {clang < 800}
```

**Comment by reneeotten** [comment]
for blacklisting of compiler you'll need to include ```PortGroup           compiler_blacklist_versions 1.0``` at the beginning of the Portfile.

Also, if you need to manually do the blacklisting (which is supposed to work correctly with the ```compiler.thread_local_storage yes```). Please do add the comments as I've shown above to reflect that so that it's clear that this is just temporarily.

**Comment by reneeotten** [comment]
> for blacklisting of compiler you'll need to include `PortGroup compiler_blacklist_versions 1.0` at the beginning of the Portfile.
 
well, not *really* at the top, that doesn't work (please do check your Portfiles with ```port lint --nitpick <port>```). Just add it after the PGs that are already present..

**Comment by reneeotten** [comment]
thanks @XNephila! For now I'll just do a "squash and merge" to have only one remaining commit.

**Pull request merged**